pipeline-templates/terraform-plan.yaml

```
@@ -3,33 +3,47 @@ jobs:
     pool:
       vmImage: ${{ parameters.agentPool }}
     steps:
-      - task: TerraformInstaller@0
+      - checkout: self
+      - checkout: cnp-azuredevops-libraries
+      - template: steps/set-build-repo-suffix-env-var.yaml@cnp-azuredevops-libraries
+      - task: Bash@3
+        displayName: Terraform install (tfenv)
         inputs:
-          terraformVersion: ${{ parameters.terraformVersion }}
+          filePath: $(System.DefaultWorkingDirectory)/cnp-azuredevops-libraries/scripts/tfenv-install-terraform.sh
       - task: TerraformCLI@0
         displayName: Init - ${{ parameters.zone }}
         inputs:
           command: 'init'
+          commandOptions: '-lock=false' # don't lock on PRs / validate phase
           backendType: 'azurerm'
           backendServiceArm: 'Reform-CFT-Mgmt'
           backendAzureRmResourceGroupName: 'reform-cft-mgmt'
           backendAzureRmStorageAccountName: 'reformcftmgmt'
           backendAzureRmContainerName: 'tfstate'
           backendAzureRmKey: 'public-dns/${{ parameters.zone }}.tfstate'
-          workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.zone }}'
+          workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.zone }}'
       - task: TerraformCLI@0
         displayName: Plan - ${{ parameters.zone }}
         inputs:
           command: 'plan'
           commandOptions: '-out=${{ parameters.zone }}${{ parameters.build }}plan -parallelism=50'
-          workingDirectory: '$(System.DefaultWorkingDirectory)/components/${{ parameters.zone }}'
+          workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.zone }}'
           environmentServiceName: 'Reform-CFT-Mgmt'
```

https://github.com/hmcts/azure-private-dns/blob/fcf04a8fa62ef123619a5e95a31571af354f432b/pipeline-templates/terraform-plan.yaml#L32
```suggestion
          environmentServiceName: 'Reform-CFT-Mgmt'
          publishPlanResults: '${{ parameters.zone }}'
```